sql/core/src/main/scala/org/apache/spark/sql/classic/DataFrameWriter.scala

```diff
@@ -484,12 +484,50 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) extends sql.DataFram
           serde = None,
           external = false,
           constraints = Seq.empty)
+        val writeOptions = if (source == "delta") {
```

Comment:
Does Apache Spark source code have this kind `delta`-specific logic before, @juliuszsompolski ?

This looks like the first proposal to have a 3rd-party company data source in Apache Spark source code. At the first glance, this string match looks a little fragile to me.